ring/ring_test.go

```diff
@@ -1632,6 +1660,19 @@ func TestRing_ShuffleShard_Consistency(t *testing.T) {
 					delete(ringDesc.Ingesters, id)
 					break
 				}
+			case enableReadOnly:
+				for id, desc := range ringDesc.Ingesters {
+					desc.ReadOnly = true
+					desc.ReadOnlyUpdatedTimestamp = time.Now().Unix()
+					ringDesc.Ingesters[id] = desc
+					break
+				}
+			case disableReadOnly:
+				for id, desc := range ringDesc.Ingesters {
+					desc.ReadOnly = false
```

Comment:
Update of `ReadOnly` field should be done together with update of `ReadOnlyUpdatedTimestamp` field.

Reply:
done in https://github.com/grafana/dskit/pull/554/commits/5d2ced6ec53289458d86768113f93fa9194b83b9.